### PR TITLE
TRUNK-5969: Wrong default directory on Windows

### PR DESCRIPTION
**Pull request by theanandankit**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5969

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/util/OpenmrsUtil.java [hunk 1]
```diff
@@ -1101,8 +1101,7 @@ public static String getApplicationDataDirectory() {
 					filepath = OpenmrsConstants.APPLICATION_DATA_DIRECTORY_FALLBACK_UNIX + File.separator + openmrsDir;
 				}
 			} else {
-				filepath = System.getProperty("user.home") + File.separator + "Application Data" + File.separator
-				        + "OpenMRS";
+				filepath = System.getenv("appdata") + File.separator + "OpenMRS";
 				if (!canWrite(new File(filepath))) {
```
Comment (ShivanshRakesh) [inline api/src/main/java/org/openmrs/util/OpenmrsUtil.java:1105]: Probably this condition is never caught in tests and hence the failing coverage check.

**Review by ShivanshRakesh — COMMENTED** [review]
LGTM!

**Comment by theanandankit** [comment]
@ShivanshRakesh thanks for review 😁

File: api/src/main/java/org/openmrs/util/OpenmrsUtil.java (hunk 1 above)
Comment (ibacher) [reply api/src/main/java/org/openmrs/util/OpenmrsUtil.java:1105]: @theanandankit First of all, thank you for this! It's important that we get this right. I would be concerned that we might be breaking existing Windows-based installs without having a proper fallback. Could we implement this as something like:

```java
filepath = System.getProperty("user.home") + File.separator + "Application Data" + File.separator + "OpenMRS";

if (!new File(filepath).exists()) { 
    filepath = System.getenv("appdata") + File.separator + "OpenMRS";
}
```

**Review by ibacher — APPROVED** [review]

**Comment by theanandankit** [comment]
updated the code as suggested by @ibacher.

**Pull request merged**